### ISSUE TITLE
false positive: prefer-const in while loop

### ISSUE THREAD

**Issue by gotofritz** [eslint/eslint#7910]
**Tell us about your environment**

* **ESLint Version: 3.13.1**
* **Node Version: 6.9.1**
* **npm Version: 3.10.8**

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
most of it is from the npm module eslint-config-pv. Prefer-const is set here
https://github.com/pro-vision/eslint-config-pv/blob/011adda61121381c6bdf4f9f3db7f979181bcfd3/rules/es6.js#L66
Below the .eslintrc

```
{
  "extends": [
    "eslint-config-pv/es6"
  ],
  "parserOptions": {
    "sourceType": "module"
  },
  "globals": {
    "jest": false
  },
  "rules": {
    "arrow-body-style": 0,
    "object-property-newline": [2, { "allowMultiplePropertiesPerLine": true }],
    "quote-props": [2, "as-needed", {
        "keywords": true,
        "unnecessary": true
      }]
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
    if (soMany > 0) {
      while (soMany--) {
        let random = randomInt(length - 1);
        // ...
      }
    }
```

**What did you expect to happen?**
let should be allowed, because the block is in a while loop so it will potentially be run several times

**What actually happened? Please include the actual, raw output from ESLint.**
I get an error `85:13  error  'random' is never reassigned. Use 'const' instead  prefer-const`

**Comment by spadgos** [comment]
`let` and `const` are block scoped, so using `const` here is actually correct.

**Comment by gotofritz** [comment]
Doh. Yes you are absolutely right, sorry for wasting your time.
